pep/app/pods/components/page/sidebar/widgets/component.ts

```diff
@@ -1,17 +1,21 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
-import { WIDGET } from 'pep/constants/sidebar';
+import { WIDGET, WidgetData } from 'pep/constants/sidebar';
 import { action } from '@ember/object';
+import { WidgetConfiguration } from 'pep/constants/configuration';
 
-interface PageSidebarWidgetsArgs {}
-
-export interface PageSidebarWidgetArgs {
+export interface PageSidebarWidgetsArgs {
+    widgets: WidgetConfiguration[];
+    data: WidgetData;
+}
+export interface PageSidebarWidgetArgs extends PageSidebarWidgetsArgs {
     toggleIsOpen: (widget: WIDGET) => void;
     openWidgets: WIDGET[];
 }
 
 export default class PageSidebarWidgets extends Component<PageSidebarWidgetsArgs> {
-    @tracked openWidgets: WIDGET[] = [WIDGET.WHATS_NEW, WIDGET.MOST_CITED, WIDGET.MOST_VIEWED];
+    @tracked openWidgets: WIDGET[] =
+        this.args.widgets?.filter((widget) => widget.open).map((widget) => widget.widget) ?? [];
```

Comment:
Can you explain your though process behind this?

Reply:
its mainly that defaulting class properties using values from instance properties (usually `args`) is always shown as done in the `constructor()` in ember docs/examples, and its a little easier for someone unfamiliar to see that this is just a one-time "setup" type logic on initial render if its done there, rather than possibly being some sort of auto-updated value that will always reflect the current `args` values.

I'm pretty sure I had also seen some potential gotchas around using this pattern at one point, though I don't remember what now.

Reply:
So Im not sure how there could be gotchas as this is just syntactic sugar for setting up properties in the constructor. Also if they are familiar with es6 classes at all, they really shouldn't be confused. If you find out if there really are gotchas I'll concede but otherwise I think this is cleaner and more straightforward.

Reply:
I thought it had something to do with the rendering lifecycle and how/when `args` are passed in, that could sometimes lead to hard to debug rendering quirks when you do something like this. I've rarely if ever seen `this.args` or even just the `this` context used to default class fields in their class body declaration, its always in the constructor.

I'll see if I can find more info on that though. So if you really don't want to change it, I would just keep an eye on it at least for any weirdness.

Reply:
Yep, i was wrong on this one 😛 so this should be safe moving forward.